### ISSUE TITLE
Validation webhook is causing over 20s POST ingresses latency

### ISSUE THREAD

**Issue by jasonBirchall** [ministryofjustice/cloud-platform#2507]
The theory is that when a single ingress rule (definition) is created, the webhook handles it in around 1s. When the number of ingress rules is over 500 the webhook takes over 15s.

To measure the validation webhook performance I will use the following promql during a performance test.
```
histogram_quantile(0.99, sum(rate(apiserver_request_duration_seconds_bucket{resource="ingresses", job="apiserver", verb!="WATCH"}[2h])) by (resource, verb, le))
```

**Comment by jasonBirchall** [comment]
Initial investigation suggests this is caused by the nginx validation webhook. To evidence this, I created a test cluster and performed the following:
- Replicating live as much as possible, a test cluster was created and simulated live traffic.
- Created over 500 applications and ingress rules.

At around 529 new ingress rules/definitions I observed the POST ingress API latency increase to over 20 seconds. This coincided with the webhook completion time rising from 1 to 13 seconds.

**Comment by jasonBirchall** [comment]
In an attempt to remedy or at least improve this latency, I performed the following on an identical test cluster (as above):
- Upgrade the Helm and image versions/tags of the ingress controller
- Upgrade the image version/tag of the nginx validation webhook.

The ingress controller upgrade showed no noticeable improvement.

The validation webhook upgrade did however improve by ~50%.

At around 529 new ingress rules/definitions I observed the POST ingress API latency increase to 13 seconds and the webhook completion time increase to 7 seconds.

Therefore, I suggest that we bump and test the webhook validation version from 1.0.0 to 1.5.0.
